packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts

```diff
@@ -99,76 +89,47 @@ export class EVMAAWallet extends LoggerWrapper {
-                            ...(usesGelatoBundler(this.chain) && gelatoBundlerProperties),
-                        });
-                        transaction = await publicClient.writeContract(request);
-                        break;
-                    }
-                    case "sft": {
-                        tokenId = evmToken.tokenId;
-                        const { request } = await publicClient.simulateContract({
-                            account: this.account,
-                            address: contractAddress,
-                            abi: erc1155,
-                            functionName: "safeTransferFrom",
-                            args: [this.getAddress(), toAddress, tokenId, (config as SFTTransferType).quantity, "0x00"],
-                            ...(usesGelatoBundler(this.chain) && gelatoBundlerProperties),
-                        });
-                        transaction = await publicClient.writeContract(request);
-                        break;
-                    }
-                    case "nft": {
-                        tokenId = evmToken.tokenId;
-                        const { request } = await publicClient.simulateContract({
-                            account: this.account,
-                            address: contractAddress,
-                            abi: erc721,
-                            functionName: "safeTransferFrom",
-                            args: [this.getAddress(), toAddress, tokenId],
-                            ...(usesGelatoBundler(this.chain) && gelatoBundlerProperties),
-                        });
-                        transaction = await publicClient.writeContract(request);
-                        break;
-                    }
-                    default: {
-                        throw new WalletSdkError(`Token not supported`);
-                    }
-                }
+            if (!isAddress(toAddress)) {
+                throw new Error(`Invalid recipient address: '${toAddress}' is not a valid EVM address.`);
+            }
 
-                if (transaction != null) {
-                    return transaction;
-                } else {
-                    throw new TransferError(
-                        `Error transferring token ${evmToken.contractAddress}
-                    ${!tokenId ? "" : ` tokenId=${tokenId}`}
-                    ${!transaction ? "" : ` with transaction hash ${transaction}`}`
-                    );
-                }
+            if (!isAddress(config.token.contractAddress)) {
+                throw new Error(
+                    `Invalid contract address: '${config.token.contractAddress}' is not a valid EVM address.`
+                );
+            }
+
+            const tx = {
+                ...transferParams({
+                    contract: config.token.contractAddress,
+                    to: toAddress,
+                    from: this.account,
+                    config,
+                }),
+            };
```

Comment:
nit: this is the same as omitting the spread, right? We can do just `const tx = transferParams(...)`?

Reply:
thanks for catching, initially I had something else in there too. Fixed